src/places/autocomplete_service.ts

```diff
@@ -1,32 +1,82 @@
 // Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 // SPDX-License-Identifier: Apache-2.0
 
-import { GeoPlacesClient, SuggestAdditionalFeature, SuggestCommand, SuggestRequest } from "@aws-sdk/client-geo-places";
+import {
+  AutocompleteCommand,
+  AutocompleteRequest,
+  GeoPlacesClient,
+  SuggestAdditionalFeature,
+  SuggestCommand,
+  SuggestRequest,
+} from "@aws-sdk/client-geo-places";
 
 import { LatLngToLngLat, MigrationLatLng, MigrationLatLngBounds, PlacesServiceStatus } from "../common";
 
-// FIXME: Temporarily add QueryAutocompletePrediction as a possible type until getPlacePredictions has been
-// re-implemented using the new Autocomplete API. All of these Autocomplete/Query predicition types can be
-// removed once we complete this work.
-interface AutocompleteResponse {
-  predictions: AutocompletePrediction[] | QueryAutocompletePrediction[];
-}
-interface AutocompletePrediction {
-  description: string;
-  place_id: string;
-}
-interface QueryAutocompletePrediction {
-  description: string;
-  matched_substrings: google.maps.places.PredictionSubstring[];
-  place_id?: string;
-  reference?: string;
-  terms: google.maps.places.PredictionTerm[];
-}
+// Handle location/bounds restrictions. bounds and location have been deprecated, and in some cases
+// have actually been removed (although still mentioned in the documentation as only deprecated).
+//   * locationBias is the top preferred field, and can be MigrationLatLng|LatLngLiteral|MigrationLatLngBounds|LatLngBoundsLiteral
+//   * bounds / locationRestriction is the next preferred field
+//   * location is the final field that is checked
+//   * radius must be paired with a LatLng (locationBias / location) to specify a circle bias
+// This logic can be shared between both getQueryPredictions and getPlacePredictions
+const setRequestLocationBias = (
+  input: AutocompleteRequest | SuggestRequest,
+  location,
```

Comment:
Most of these variables don't have their type set in the syntax and only one has it mentioned in the comment above.  If there a reason we have these as one function instead of splitting them into separate polymorphic functions? The deprecation discussion in the comment is a little confusing as well.

Reply:
Okay I see below that this is moving existing logic into a sub function, not exposing this function for an API so splitting this into separate functions would leave a lot of if/else logic in the original function.

Reply:
Okay I see below that this is moving existing logic into a sub function, not exposing this function for an API so splitting this into separate functions would leave a lot of if/else logic in the original function.